modules/core/src/layers/xviz-layer.js

```diff
@@ -259,32 +302,40 @@ export default class XVIZLayer extends CompositeLayer {
     return null;
   }
 
-  updateState({props, oldProps, changeFlags}) {
-    let {type} = this.state;
+  updateState({ props, oldProps, changeFlags }) {
+    let { type } = this.state;
 
     if (changeFlags.dataChanged) {
       // Pre-process data
       let data = props.data;
       const dataType = this._getLayerType(data);
       type = XVIZ_TO_LAYER_TYPE[dataType];
 
-      if (type === 'scatterplot' && data[0].vertices && Array.isArray(data[0].vertices[0])) {
+      if (
+        !data.vertexBuffer &&
```

Comment:
What is this vertexBuffer?  it's related other work but I don't think it exists here yet correct?